lib/elixir/lib/macro.ex

```diff
@@ -2376,14 +2385,19 @@ defmodule Macro do
 
   """
   @doc since: "1.14.0"
-  @spec inspect_atom(:literal | :key | :remote_call, atom) :: binary
-  def inspect_atom(source_format, atom)
+  @spec inspect_atom(:literal | :key | :remote_call, atom, keyword) :: binary
+  def inspect_atom(source_format, atom, opts \\ []) do
+    opts = Keyword.validate!(opts, [:escape])
 
-  def inspect_atom(:literal, atom) when is_nil(atom) or is_boolean(atom) do
+    escape = Keyword.get(opts, :escape, &inspect_atom_escape/2)
```

Comment:
I would try to get the function from options only if we need to escape. Something like `escape(opts, binary, ?")` down below.